### PR TITLE
[clang-format] Fix regression with BlockIndent of Braced Initializers

### LINKED ISSUE #73584: clang-format: Behavior changes between 16.0 and 17.0
Hi, when I update the clang-format from v16.0 to v17.0, the behavior of the following code block changes:
Before (16.0):
```
    PyObjectPtr<PyObject> const readinto_method_obj{
            PyObject_GetAttrString(input_stream, "readinto")};
```
After (17.0):
```
    PyObjectPtr<PyObject> const readinto_method_obj{PyObject_GetAttrString(input_stream, "readinto")
    };
```
Is this an expected behavior? Is there any new settings I need to update to keep the old behavior, or at least sth like:
```
    PyObjectPtr<PyObject> const readinto_method_obj{
            PyObject_GetAttrString(input_stream, "readinto")
    };
```
Here is my clang-format config file: https://github.com/y-scope/clp-ffi-py/blob/main/.clang-format
Thanks!

### PR DESCRIPTION
**Pull request by gedare**
Fixes #73584.

### CODE REVIEW

**Review by mydeveloperday — APPROVED** [review]
This looks ok to me can we have @owenca or @HazardyKnusperkeks  give it the once over, but I'm good with it.

File: clang/unittests/Format/FormatTest.cpp [hunk 1]
```diff
@@ -9336,6 +9336,9 @@ TEST_F(FormatTest, AlignsAfterOpenBracket) {
                "    ccccccc(aaaaaaaaaaaaaaaaa,         //\n"
                "        b));",
                Style);
+  verifyFormat("aaaaaaa<bbbbbbbb> const aaaaaaaaaa{\n"
+               "    aaaaaaaaaaaaa(aaaaaaaaaaa, aaaaaaaaaaaaaaaa)};",
+               Style);
```
Comment (owenca) [inline clang/unittests/Format/FormatTest.cpp:9341]: This test seems irrelevant and IMO ought to be deleted.

File: clang/unittests/Format/FormatTest.cpp [hunk 2]
```diff
@@ -9395,6 +9398,9 @@ TEST_F(FormatTest, AlignsAfterOpenBracket) {
       "fooooooooooo(new FOO::BARRRR(\n"
       "    XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXZZZZZZZZZZZZZZZZZZZZZZZZZ()));",
       Style);
+  verifyFormat("aaaaaaa<bbbbbbbb> const aaaaaaaaaa{\n"
+               "    aaaaaaaaaaaaa(aaaaaaaaaaa, aaaaaaaaaaaaaaaa)};",
+               Style);
```
Comment (owenca) [inline clang/unittests/Format/FormatTest.cpp:9403]: Ditto.

File: clang/lib/Format/ContinuationIndenter.cpp [hunk 3]
```diff
@@ -348,6 +348,13 @@ bool ContinuationIndenter::canBreak(const LineState &State) {
     }
   }
 
+  // Don't allow breaking before a closing right brace of a block-indented
+  // braced list initializer if there was not already a break.
```
Comment (owenca) [inline clang/lib/Format/ContinuationIndenter.cpp:352]:
```suggestion
  // Don't allow breaking before a closing brace of a block-indented braced list
  // initializer if there isn't already a break.
```

**Review by owenca — APPROVED** [review]

**Pull request merged**